cpp/include/cudf/detail/gather.cuh

```diff
@@ -265,18 +272,19 @@ gather(table_view const& source_table, MapIterator gather_map_begin,
   auto source_table_view = table_device_view::create(source_table);
   auto destination_table_view = mutable_table_device_view::create(destination_table->mutable_view());
 
-  bitmask_kernel<<<gather_grid_size, gather_block_size>>>(
-                                                          *source_table_view,
+  bitmask_kernel<<<gather_grid_size, gather_block_size, 0, stream>>>(*source_table_view,
                                                           gather_map_begin,
-                                                          *destination_table_view);
-
+                                                          *destination_table_view,
+                                                          valid_counts.data().get());
 
-  mutable_table_view dest_view = destination_table->mutable_view();
+  CUDA_TRY(cudaMemcpy(h_valid_counts.data(), valid_counts.data().get(),
```

Comment:
If you use a `thrust::host_vector<size_type>` instead of a `std::vector<size_type>` this copy can be easier:
```
thrust::host_vector<size_type> h_valid_counts(valid_counts);
```
It will automatically copy device to host.